graph-proxy/src/graphql/workflows.rs

```diff
@@ -318,6 +368,22 @@ impl Task {
     async fn dependencies(&self) -> Vec<String> {
         self.node_status.children.clone()
     }
+
+    /// Artifacts produced by a task
+    async fn artifacts(&self) -> Vec<Artifact> {
+        match self.node_status.outputs.as_ref() {
+            None => Vec::new(),
+            Some(outputs) => outputs
+                .artifacts
+                .iter()
+                .map(|manifest| Artifact {
+                    manifest,
+                    metadata: self.metadata,
+                    node_id: &self.node_status.id,
+                })
+                .collect::<Vec<_>>(),
+        }
```

Comment:
I might be tempted to `map` here

```suggestion
        self.node_status.outputs.as_ref().map(|outputs| {
            outputs
                .artifacts
                .iter()
                .map(|manifest| Artifact {
                    manifest,
                    metadata: self.metadata,
                    node_id: &self.node_status.id,
                })
                .collect::<Vec<_>>()
        }).unwrap_or_default()
```

Reply:
I like it, done!